app/controllers/companies_controller.rb

```diff
@@ -38,6 +38,14 @@ def destroy
 
   private
 
+  def  companies_query
+    if current_user.company
+      Company.accessible_by(current_ability)
+    else
+      Company.all
+    end
+  end
+
```

Comment:
`query = current_user.company ? Company.accessible_by(current_ability) : Company.all`
Maybe like this? to avoid extra methods with if else condition?